### PR TITLE
KAFKA-7183: Add a trogdor test that creates many connections to brokers

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/trogdor/workload/ConnectStressSpec.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The specification for a task which connects and disconnects many times a
+ * second to stress the broker.
+ */
+public class ConnectStressSpec extends TaskSpec {
```
Comment (ijuma) [inline tools/src/main/java/org/apache/kafka/trogdor/workload/ConnectStressSpec.java]: Maybe `ConnectionStressSpec`? The current name sounds like it's a `Connect` test.

**Comment by ijuma** [comment]
@rajinisivaram, do you have time to review this PR?

File: tools/src/main/java/org/apache/kafka/trogdor/workload/Throttle.java [hunk 2]
```diff
@@ -52,6 +54,10 @@ synchronized public boolean increment() throws InterruptedException {
         }
     }
 
+    public synchronized long lastNow() {
```
Comment (rajinisivaram) [inline tools/src/main/java/org/apache/kafka/trogdor/workload/Throttle.java]: Odd name for the method - could we rename to indicate what the time refers to?

**Review by rajinisivaram — COMMENTED** [review]
@cmccabe Thanks for the PR, LGTM. Left a minor comment, also agree with @ijuma that it would be better to use `Connection` instead of `Connect`.

File: tools/src/main/java/org/apache/kafka/trogdor/workload/Throttle.java (hunk 2 above)
Comment (cmccabe) [reply tools/src/main/java/org/apache/kafka/trogdor/workload/Throttle.java]: I renamed it to lastTimeMs

**Comment by ijuma** [comment]
@rajinisivaram is this ready to be merged?

**Review by rajinisivaram — APPROVED** [review]
@cmccabe Thanks for the updates, LGTM. Merging to trunk.

**Pull request merged**